src/main/java/pl/kinga/treatment/user/UserService.java

```diff
@@ -0,0 +1,33 @@
+package pl.kinga.treatment.user;
+
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import pl.kinga.treatment.user.User;
+import pl.kinga.treatment.user.UserRepository;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+@Service
+public class UserService {
+    private final PasswordEncoder passwordEncoder;
+    private final UserRepository userRepository;
+
+    public UserService(PasswordEncoder passwordEncoder, UserRepository userRepository) {
+        this.passwordEncoder = passwordEncoder;
+        this.userRepository = userRepository;
+    }
+
+    public void registerUser(String userName, String rawPassword) {
+
+        User userToAdd = new User();
+        userToAdd.setUserName(userName);
+        String encryptedPassword = passwordEncoder.encode(rawPassword);
+        List<UserRole> list = Arrays.asList(new UserRole(Role.ROLE_USER, userToAdd));
+        userToAdd.setRoles(new HashSet<>(list));
```

Comment:
W javie9 stworzono metode Set.of, więc zamiast stworzyć najpierw listę a potem z listy set, to powinno zadziałać takie coś:
`userToAdd.setRoles(Set.of(new UserRole(Role.ROLE_USER, userToAdd)));`